modules/systemd-journal/journal-reader.c

```diff
@@ -598,7 +598,7 @@ _init(LogPipe *s)
   if (!log_source_init(s))
     return FALSE;
 
-  gint res = journald_open(self->journal, SD_JOURNAL_LOCAL_ONLY);
+  gint res = journald_open(self->journal, SD_JOURNAL_LOCAL_ONLY | SD_JOURNAL_ALL_NAMESPACES);
```

Comment:
According to the man page:

```
       sd_journal_open() opens the log journal for reading. It will find all journal files automatically and interleave them automatically when reading. As first argument it takes a
       pointer to a sd_journal pointer, which, on success, will contain a journal context object. The second argument is a flags field, which may consist of the following flags ORed
       together: SD_JOURNAL_LOCAL_ONLY makes sure only journal files generated on the local machine will be opened.  SD_JOURNAL_RUNTIME_ONLY makes sure only volatile journal files will
       be opened, excluding those which are stored on persistent storage.  SD_JOURNAL_SYSTEM will cause journal files of system services and the kernel (in opposition to user session
       processes) to be opened.  SD_JOURNAL_CURRENT_USER will cause journal files of the current user to be opened. If neither SD_JOURNAL_SYSTEM nor SD_JOURNAL_CURRENT_USER are
       specified, all journal file types will be opened.

       sd_journal_open_namespace() is similar to sd_journal_open() but takes an additional namespace parameter that specifies which journal namespace to operate on. If specified as NULL
       the call is identical to sd_journal_open(). If non-NULL only data from the namespace identified by the specified parameter is accessed. This call understands two additional flags:
       if SD_JOURNAL_ALL_NAMESPACES is specified the namespace parameter is ignored and all defined namespaces are accessed simultaneously; if SD_JOURNAL_INCLUDE_DEFAULT_NAMESPACE the
       specified namespace and the default namespace are accessed but no others (this flag has no effect when namespace is passed as NULL). For details about journal namespaces see
       systemd-journald.service(8).
```

The `sd_journal_open` does not handle this flag according to the man page. I would either consider moving to the new call (when available) or get it confirmed that this is a documentation bug.

Reply:
`sd_journal_open` is defined as:

https://github.com/systemd/systemd/blob/12d14b73732df8c3ca46383573eeea069832969f/src/journal/sd-journal.c#L1971-L1973

So it will naturally allow this flag. Nevertheless, I agree that using the new function is a better way and circumvents all documentation ambiguities.